bdb/cursor.c

```diff
@@ -2587,7 +2584,7 @@ static int bdb_update_pglogs_fileid_queues(
         }
 
         // Sanity
-        if (chk = fileid_queue->queue_keys.bot)
+        if ((chk = fileid_queue->queue_keys.bot))
```

Comment:
Although this shuts up the compiler just as well, I prefer exlicit check e.g. `if ((chk = fileid_queue->queue_keys.bot) != 0)` for added readability (like berk code does).

Reply:
I'm actually ambivalent about which is more readable: 
 If (rc) 
Or
 If (rc != 0)

As such I think I will initially refrain from changing it one way or the other (there are  instances of the other in this PR). But if we agree on making such change, I can certainly do so.

Reply:
Please don't write this as ((a=b)).  Please either use a=b;if(a), or  if((a=n)!=0).